addons/udes_delivery_control/tests/test_related_pickings.py

```diff
@@ -0,0 +1,158 @@
+                    expected_pick_names = expected_picks.mapped("name")
+
+                    self.assertEqual(
+                        returned_picks,
+                        expected_picks,
+                        f"'{field}' for {picking.name} should be '{expected_pick_names}'",
+                    )
+
+    def test_assert_delivery_control_goods_in_relationship(self):
+        """
+        Assert that the first/previous/next picking fields are computed correctly
+
+        Delivery Control should be the first picking and before the Goods In picking
+        """
+        expected_pickings_by_field = {
+            "u_first_picking_ids": {
+                self.delivery_control_picking: self.delivery_control_picking,
+                self.goods_in_picking: self.delivery_control_picking,
+            },
+            "u_prev_picking_ids": {
+                self.delivery_control_picking: False,
+                self.goods_in_picking: self.delivery_control_picking,
+            },
+            "u_next_picking_ids": {
+                self.delivery_control_picking: self.goods_in_picking,
+                self.goods_in_picking: False
+            },
+        }
+
+        picks = self.delivery_control_picking | self.goods_in_picking
+
+        # Assert that each computed picking field returns the expected result for all picks
+        self._assert_picking_related_pickings_match_expected_values(
+            picks, expected_pickings_by_field
+        )
+
+    def test_assert_related_pickings_computed_correctly_with_additional_pickings(self):
+        """
+        Assert that the first/previous/next picking fields are computed correctly
+        with the following setup in addition to the Delivery Control and Goods In picking:
+
+        3 picks (A, B and C):
+
+        - Pick A - direct parent of Pick C
+        - Pick B - originates from Goods In Pick
+        - Pick C - originates from Goods In Pick and Pick A
+        """
+        # Create extra picks
+        pick_a = self.create_picking(self.picking_type_pick, name="Pick A")
+        pick_b = self.create_picking(self.picking_type_pick, name="Pick B")
+        pick_c = self.create_picking(self.picking_type_pick, name="Pick C")
+
+        # Create moves for Goods In and extra picks
+        goods_in_move = self.create_move(self.apple, 1, self.goods_in_picking)
+        pick_a_move = self.create_move(self.apple, 1, pick_a)
+        pick_b_move = self.create_move(self.apple, 1, pick_b)
+        pick_c_move = self.create_move(self.apple, 1, pick_c)
+
+        # Set Pick B's move to have originated from Goods In's move
+        pick_b_move.move_orig_ids = goods_in_move
+
+        # Set Pick C's move to have originated from Goods In and Pick A's move
+        pick_c_move.move_orig_ids = goods_in_move | pick_a_move
+
+        # Pick Relationship Diagram
+        #
+        # DC         A
+        # |          |
+        # GI____     |
+        # |     |    |
```

Comment:
nice ascii art, but maybe could use \ and / above C.
It looks like GI comes from the top but A from the bottom

Reply:
Art in general has never been my strong point!

But updated now with your suggestion. :)